core/src/main/java/uk/gov/gchq/koryphe/impl/function/ToIterable.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2022 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.gchq.koryphe.impl.function;
+
+
+import uk.gov.gchq.koryphe.Since;
+import uk.gov.gchq.koryphe.Summary;
+import uk.gov.gchq.koryphe.function.KorypheFunction;
+
+import java.util.Arrays;
+import java.util.Collections;
+
+@Since("2.5.0")
+@Summary("Converts an object into a Iterable")
+public class ToIterable extends KorypheFunction<Object, Iterable<Object>> {
+    public ToIterable() {
+    }
+
+    public Iterable<Object> apply(final Object value) {
+        final Iterable<Object> rtn;
+        if (null == value) {
+            rtn = Collections.emptyList();
+        } else if (value instanceof Iterable) {
+            //noinspection unchecked
+            rtn = (Iterable<Object>) value;
```

Comment:
This line is key, if the input is a lazy iterable the same iterable it returned, whereas in `ToList` the whole iterable has to be read to make a list

Reply:
Wouldn't it be possible to change `ToList` to have a flag to do this instead of a duplicating class?
I know there was a reason Koryphe is being used somewhere like this, but if we know we have an `Iterable`, it doesn't make sense to me to be using Koryphe to turn something which is already an `Iterable` into itself. So while `ToList` is not a good solution, I'm not sure of the reason for using Koryphe at all.
It makes sense for Koryphe to accept a list and return the same thing, but I don't see why anyone would do this unless they didn't know they already had a list.

Reply:
You don't know what is being returned from the individual graphs, until its returned.

3 graphs could return:
3 nulls
3 iterates of elements
3 schemas
3 strings
3 lists of strings.